src/libraries/System.Net.Primitives/ref/System.Net.Primitives.cs

```diff
@@ -323,6 +323,23 @@ public abstract partial class TransportContext
         protected TransportContext() { }
         public abstract System.Security.Authentication.ExtendedProtection.ChannelBinding? GetChannelBinding(System.Security.Authentication.ExtendedProtection.ChannelBindingKind kind);
     }
+    public enum NetworkError : int
+    {
+        Unknown = 0,
+        AddressInUse,
+        InvalidAddress,
+        HostNotFound,
+        ConnectionRefused,
+        ConnectionAborted,
+        ConnectionReset,
+    }
+    public class NetworkException : System.IO.IOException
+    {
+        protected NetworkException(NetworkError error, Exception? innerException = null, string? message = null) { }
```

Comment:
This order is strange. message (almost?) always comes before innerException.

Reply:
They're ordered from most used to least used.

Reply:
That may be, but where else in the dozens or hundreds of exception types in .NET does this order occur? It's very inconsistent. @bartonjs, you're ok with this?

Reply:
The order from the ctor ended up just being what we designed the factory method as; which was based on the notion that it would be `throw NetworkException.Create(NetworkError.ConnectionAborted);`, which would allow the easy creation of a derived type later, to avoid using exception filters as the primary means of distinguishing errors that seem catchable.

I don't think we really spent a lot of time and thinking on this line, TBH; and @geoffkizer was pushing for a fast "can we just approve it and move on", which we apparently did :smile:.

Reply:
>  @geoffkizer was pushing for a fast "can we just approve it and move on"

I plead guilty to this :)

Reply:
By far the most common usage here is, give an error code and an optional inner exception. It's pretty weird to want to customize the error message. But I don't think we should rule it out entirely -- especially for the NetworkError.Unknown case, it's useful for the provider to be able to provide more info in the exception message.

Perhaps we could finesse this by having two overloads:
* NetworkException(NetworkError error, Exception? innerException = null)
* NetworkException(string message, NetworkError error, Exception? innerException = null)